codex/blockexchange/engine/engine.nim

```diff
@@ -309,24 +301,22 @@ proc blocksHandler*(
     peerCtx = b.peers.get(peer)
 
   if peerCtx != nil:
-    # we don't care about this blocks anymore, lets cleanup the list
     await b.payForBlocks(peerCtx, blocks)
+    ## shouldn't we remove them from the want-list instead of this:
```

Comment:
This is handled in `resolveBlocks`, not really sure if we want `cleanPresence` is doing there, it doesn't seem like it's needed and it's probably breaking something.